sandbox/lua/encoders/es_payload.lua

```diff
@@ -72,8 +72,16 @@ function process_message()
     if ts_from_message then
         ns = read_message("Timestamp")
     end
+
     local idx_json = elasticsearch.bulkapi_index_json(index, type_name, id, ns)
-    add_to_payload(idx_json, "\n", read_message("Payload"))
+    local payload  = read_message("Payload")
+
+    -- bulk api expects newline in the end of the payload
+    if not string.match(payload, "\n$") then
+        add_to_payload("\n")
+    end
+
+    add_to_payload(idx_json, "\n", payload)
```

Comment:
You actually put this in the wrong order, we want the final "\n" to be added to the payload *after* the first batch of data. I've fixed it up and will merge, however.